disnake/ext/commands/interaction_bot_base.py

```diff
@@ -823,14 +995,44 @@ async def _prepare_application_commands(self) -> None:
         await self.wait_until_first_connect()
         await self._cache_application_commands()
         await self._sync_application_commands()
+        self._fill_app_command_ids()
         self._sync_queued = False
 
+    def _fill_app_command_ids(self):
+        # this requires that _cache_application_commands was ran first
+        if not isinstance(self, disnake.Client):
+            raise NotImplementedError("This method is only usable in disnake.Client subclasses")
+
+        all_commands = self._all_app_commands
+        for api_command in self._connection._global_application_commands.values():
+            cmdmeta = disnake.utils.get(all_commands, name=api_command.name, guild_id=None)
```

Comment:
Again, we can speed this up by constructing meta first and then checking if it's in the dict:
```python
cmdmeta = AppCommandMetadata(api_command.name, api_command.type, None)
cmd = all_commands.get(cmdmeta)
if cmd is None:
    continue
```